package.json

```diff
@@ -43,15 +47,17 @@
       "nonbsp": false
    },
    "scripts": {
-      "pretest": "jshint *.js spec docs && gulp lint-html && gulp build-dist",
-      "test": "mocha --timeout 10000 spec/*.js"
+      "pretest": "jshint *.js spec docs && gulp lint-html && npm run build",
+      "test": "mocha --timeout 10000 spec/*.js",
+      "build": "tsc && gulp build-dist",
+      "typecheck": "tsc --noEmit"
```

Comment:
When does the `"typecheck"` script get called?

Reply:
You can call it manually. The `build` task calls the same `tsc` which checks the typing and errors out, so gulp is not called afterwards. I didn't see the point in having a `prebuild` if they don't do something different.

What do you think?